### PR TITLE
IntegrateEllipsoids after PredictPeaks

### LINKED ISSUE #25739: bug in IntegrateEllipsoids for predicted  peaks
### Expected behavior
IntegrateEllipsoids works
### Actual behavior
"Runtime Error: At least three linearly independent indexed peaks are needed" during IntegrateEllipsoids
### Steps to reproduce the behavior
Run SCD reduction with PredictPeaks and IntegrateEllipsoids

### PR DESCRIPTION
**Pull request by VickieLynch**
**Description of work.**
Add line to PredictPeaks so IntegrateEllipsoids works for predicted peaks. This is related to modulated structure changes, but effects all peaks.

**To test:**
Use TOPAZ_3132 in SystemTestData:
````python
Load(Filename='TOPAZ_3132_event.nxs', OutputWorkspace='TOPAZ_3132_event', FilterByTofMin=500, FilterByTofMax=16666)
FilterBadPulses(InputWorkspace='TOPAZ_3132_event', OutputWorkspace='TOPAZ_3132_event', LowerCutoff=25)
ConvertToMD(InputWorkspace='TOPAZ_3132_event', QDimensions='Q3D', dEAnalysisMode='Elastic', Q3DFrames='Q_sample', LorentzCorrection=True, OutputWorkspace='TOPAZ_3132_md', MinValues='-25,-25,-25', MaxValues='25,25,25', SplitInto='2', SplitThreshold=50, MaxRecursionDepth=13, MinRecursionDepth=7)
FindPeaksMD(InputWorkspace='TOPAZ_3132_md', PeakDistanceThreshold=0.37680000000000002, MaxPeaks=50, DensityThresholdFactor=100, OutputWorkspace='TOPAZ_3132_peaks')
FindUBUsingFFT(PeaksWorkspace='TOPAZ_3132_peaks', MinD=3, MaxD=15, Tolerance=0.12)
CopySample(InputWorkspace='TOPAZ_3132_peaks', OutputWorkspace='TOPAZ_3132_event', CopyName=False, CopyMaterial=False, CopyEnvironment=False, CopyShape=False)
IndexPeaks(PeaksWorkspace='TOPAZ_3132_peaks', Tolerance=0.12)
PredictPeaks(InputWorkspace='TOPAZ_3132_peaks', WavelengthMin=0.40000000000000002, WavelengthMax=3.5, MinDSpacing=0.40000000000000002, MaxDSpacing=8.5, OutputWorkspace='TOPAZ_3132_peaks')
IntegrateEllipsoids(InputWorkspace='TOPAZ_3132_event', PeaksWorkspace='TOPAZ_3132_peaks', RegionRadius=0.25, SpecifySize=True, OutputWorkspace='TOPAZ_3132_peaks')
````

Fixes #25739

*This does not require release notes* because **this bug was never in a release*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by wdzhou — APPROVED** [review]
This looks good to me.

:shipit: after all the tests are passed.

File: Framework/Crystal/src/PredictPeaks.cpp [hunk 1]
```diff
@@ -627,6 +627,7 @@ void PredictPeaks::calculateQAndAddToOutput(const V3D &hkl,
   // Save the run number found before.
   peak->setRunNumber(m_runNumber);
   peak->setHKL(hkl * m_qConventionFactor);
+  peak->setIntHKL(hkl * m_qConventionFactor);
```
Comment (quantumsteve) [inline Framework/Crystal/src/PredictPeaks.cpp]: Would it make sense for `Peaks::setHKL` to call `Peaks::setIntHKL` internally or vice versa? Or is there a case where one would want them to different?

Comment (VickieLynch) [reply Framework/Crystal/src/PredictPeaks.cpp]: @quantumsteve Some of the modulation vectors are 0.5 and above, so we cannot round the HKL and always get the correct int HKL.

Comment (quantumsteve) [reply Framework/Crystal/src/PredictPeaks.cpp]: :+1:

**Pull request merged**